mlflow/utils/validation.py

```diff
@@ -367,6 +367,17 @@ def _validate_experiment_id_type(experiment_id):
 def _validate_model_name(model_name):
     if model_name is None or model_name == "":
         raise MlflowException("Registered model name cannot be empty.", INVALID_PARAMETER_VALUE)
+    elif not isinstance(model_name, str):
+        raise MlflowException(
+            f"Invalid type: Registered model name expected to be a str, got {type(model_name)}",
+            INVALID_PARAMETER_VALUE,
+        )
+
+    validator_regex = r"^[^;:\/? \$&+,=@\"<>#%]+$"
```

Comment:
can we only check for `/`, `\` and `:` for now? as we don't want to break any backward compatibility